x-pack/plugins/advanced_ui_actions/public/components/action_wizard/test_data.tsx

```diff
@@ -0,0 +1,173 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiFieldText, EuiFormRow, EuiSelect, EuiSwitch } from '@elastic/eui';
+import { ActionFactory, ActionBaseConfig, ActionWizard } from './action_wizard';
+
+export const dashboards = [
+  { id: 'dashboard1', title: 'Dashboard 1' },
+  { id: 'dashboard2', title: 'Dashboard 2' },
+];
+
+export const dashboardDrilldownActionFactory: ActionFactory<{
+  dashboardId?: string;
+  useCurrentDashboardFilters: boolean;
+  useCurrentDashboardDataRange: boolean;
+}> = {
+  type: 'Dashboard',
+  displayName: 'Go to Dashboard',
+  iconType: 'dashboardApp',
+  createConfig: () => {
+    return {
+      dashboardId: undefined,
+      useCurrentDashboardDataRange: true,
+      useCurrentDashboardFilters: true,
+    };
+  },
+  isValid: (name, config) => {
+    if (!name) return false;
+    if (!config.dashboardId) return false;
+    return true;
+  },
+  wizard: props => {
+    const config = props.config ?? {
+      dashboardId: undefined,
+      useCurrentDashboardDataRange: true,
+      useCurrentDashboardFilters: true,
+    };
+    return (
+      <>
+        <EuiFormRow label="Choose destination dashboard:">
+          <EuiSelect
+            name="selectDashboard"
+            hasNoInitialSelection={true}
+            options={dashboards.map(({ id, title }) => ({ value: id, text: title }))}
+            value={config.dashboardId}
+            onChange={e => {
+              props.onConfig({ ...config, dashboardId: e.target.value });
+            }}
+          />
+        </EuiFormRow>
+        <EuiFormRow hasChildLabel={false}>
+          <EuiSwitch
+            name="useCurrentFilters"
+            label="Use current dashboard's filters"
+            checked={config.useCurrentDashboardFilters}
+            onChange={() =>
+              props.onConfig({
+                ...config,
+                useCurrentDashboardFilters: !config.useCurrentDashboardFilters,
+              })
+            }
+          />
+        </EuiFormRow>
+        <EuiFormRow hasChildLabel={false}>
+          <EuiSwitch
+            name="useCurrentDateRange"
+            label="Use current dashboard's date range"
+            checked={config.useCurrentDashboardDataRange}
+            onChange={() =>
+              props.onConfig({
+                ...config,
+                useCurrentDashboardDataRange: !config.useCurrentDashboardDataRange,
+              })
+            }
+          />
+        </EuiFormRow>
+      </>
+    );
+  },
+};
+
+export const urlDrilldownActionFactory: ActionFactory<{ url: string; openInNewTab: boolean }> = {
+  type: 'Url',
+  displayName: 'Go to URL',
+  iconType: 'link',
+  createConfig: () => {
+    return {
+      url: '',
+      openInNewTab: false,
+    };
+  },
+  isValid: (name, config) => {
+    if (!name) return false;
```

Comment:
nit: I'd probably just pass in config here.  No factory should allow an empty name so the check should probably be external rather than giving control to the individual factory implementation.

Reply:
I was thinking of the use case with pattern. e.g. check if this is email or url.

Reply:
If an action needed an email address or url, I think it would be part of `config`, not the name?  Factory wizard also has no control to highlight the name input field with an error, like it does with its own form controls.

Reply:
Oh, right, completely agree with you. will change it. 
(I guess I confused myself with action factory wizards where we had url)

Reply:
improved